src/stream.cpp

```diff
@@ -161,7 +161,7 @@ namespace oxen::quic
     void Stream::send_impl(bstring_view data, std::shared_ptr<void> keep_alive)
     {
         if (data.empty())
-            throw std::invalid_argument{"Cannot send empty byte string"};
```

Comment:
I honestly kinda like that this throws on an empty send? It immediately tells us that something is wrong in whatever application logic is trying to feed data to libquic

Though I've been off in refactor land for a long time, def gonna pick your brain off band on this

Reply:
> I honestly kinda like that this throws on an empty send? It immediately tells us that something is wrong in whatever application logic is trying to feed data to libquic

The thing is, it only *might* be a bug in application logic, but by throwing here we're forcing it to be an application bug.  It's kind of like `std::cout << "";` -- it does nothing, but isn't necessarily a bug.

Reply:
Case in point: my echo-server broke without this because it reads a line of input, then sends it, and broke when I hit enter without typing anything.  It feels weird for that to be an invalid use of `send()` and for me to have to make sure that anywhere I call send() I've already checked for non-emptiness.

Reply:
Shit. Valid.